yasmin_ros/include/yasmin_ros/action_state.hpp

```diff
@@ -70,7 +70,7 @@ template <typename ActionT> class ActionState : public yasmin::State {
       : ActionState(nullptr, action_name, create_goal_handler, outcomes,
                     result_handler, feedback_handler, timeout) {}
 
-  ActionState(rclcpp::Node *node, std::string action_name,
+  ActionState(const rclcpp::Node::SharedPtr &node, std::string action_name,
```

Comment:
Curious bystander here:
Why do you pass a reference of a SharedPtr? It's not cheaper than actually passing the pointer is it?

Reply:
Hello,
Yes, i have implemented it because of performance optimization. Passing by value here is unnecessary stuff because of the fact that you already use assign operation here.

https://github.com/uleroboticsgroup/yasmin/blob/740dcafbba941114159e0a4be0102ef8e890b681/yasmin_ros/include/yasmin_ros/action_state.hpp#L93-L97

https://github.com/uleroboticsgroup/yasmin/blob/740dcafbba941114159e0a4be0102ef8e890b681/yasmin_ros/include/yasmin_ros/action_state.hpp#L175